File: JAVA/S1_BASIC/L5_BASICRECURSION/ntoone.java
class recursion{
    void nto1(int n){
        if(n==0)return;
        System.out.print(n+" ");
        nto1(n-1);

    }
    
}
public class ntoone{
    public static void main(String[] args){
        recursion r=new recursion();
        r.nto1(5);
    }
}

File: python/S1_BASIC/L5_BASICRECURSION/ntoone.py
class Recursion:
    def __init__(self,n):
        self.n=n
    def ntone(self,n):
        if n==0:
            return 
        print(n,end=" ")
        self.ntone(n-1)
if __name__=="__main__":
    n=int(input())
    R=Recursion(n)
    R.ntone(R.n)
        